File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        HashMap<Integer, Integer> hMap=new HashMap<>();
        for(int i=0;i<nums.length;i++){
            if(!hMap.containsKey(nums[i])){
                hMap.put(nums[i], i);
            }else{
                int value=hMap.get(nums[i]);
                if(Math.abs(value-i)<=k){
                    return true;
                }else{
                    hMap.put(nums[i], i);
                }
            }
        }
        return false;
    }
}

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        hset=set()

        l=0
        for r in range(len(nums)):
            if nums[r] in hset:
                return True
            
            hset.add(nums[r])
            if (r-l+1)>k:
                hset.remove(nums[l])
                l+=1
        
        return False